homeassistant/components/weatherflow_cloud/config_flow.py

```diff
@@ -0,0 +1,61 @@
+"""Config flow for WeatherflowCloud integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from aiohttp import ClientResponseError
+import voluptuous as vol
+from weatherflow4py.api import WeatherFlowRestAPI
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+
+async def _async_validate_api_token(api_token) -> bool:
+    """Validate the API token."""
+
+    async with WeatherFlowRestAPI(api_token) as api:
+        await api.async_get_stations()
+
+    return True
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WeatherFlowCloud."""
+
+    VERSION = 1
+
+    async def async_step_user(
```

Comment:
Maybe flipping the logic in this step makes it more readable and easier to manage, (So from `if user_input is None:` to `if user_input is not None:`)